baseapp/baseapp.go

```diff
@@ -379,7 +379,7 @@ func (app *BaseApp) BeginBlock(req abci.RequestBeginBlock) (res abci.ResponseBeg
 	} else {
 		// In the first block, app.deliverState.ctx will already be initialized
 		// by InitChain. Context is now updated with Header information.
-		app.deliverState.ctx = app.deliverState.ctx.WithBlockHeader(req.Header)
+		app.deliverState.ctx = app.deliverState.ctx.WithBlockHeader(req.Header).WithBlockHeight(req.Header.Height)
```

Comment:
Should this be fixed on develop through a seperate PR?

Reply:
Maybe, it's minor and wasn't caught before since we only hit this case on height 0

Reply:
Added to the changelog here for now